packages/data/src/registry.js

```diff
@@ -211,10 +211,18 @@ export function createRegistry( storeConfigs = {}, parent = null ) {
 	/**
 	 * Registers a store instance.
 	 *
-	 * @param {string} name  Store registry name.
-	 * @param {Object} store Store instance object (getSelectors, getActions, subscribe).
+	 * @param {string}   name        Store registry name.
+	 * @param {Function} createStore Function that creates a store object (getSelectors, getActions, subscribe).
 	 */
-	function registerStoreInstance( name, store ) {
+	function registerStoreInstance( name, createStore ) {
+		if ( stores[ name ] ) {
+			// eslint-disable-next-line no-console
+			console.error( 'Store "' + name + '" is already registered.' );
```

Comment:
Should this be an info or warning call instead? IMO, we want this to be a supported option for stores